File: examples/biased-coin-flip/prior-sampler.py
from numpy.random import uniform
from sys import argv, stdout, stderr

# Process arguments
if len(argv) != 3:
    stderr.write("Usage: {} Q_LOW Q_HIGH\n"
            "Sample from a uniform distribution with "
            "lower bound Q_LOW and upper bound Q_HIGH\n".format(argv[0]))

    exit(1)

q_low = float(argv[1])
q_high = float(argv[2])

# Sample from uniform distribution
q_sampled = uniform(low=q_low, high=q_high)

# Print sampled parameter
stdout.write("{}\n".format(q_sampled))

File: examples/biased-coin-flip/prior-sampler.cc
#include <string>
#include <iostream>
#include <chrono>
#include <random>

int main(int argc, char *argv[])
{
    // Process arguments
    if (argc != 3)
    {
        std::cerr << "Usage: " << argv[0] << " Q_LOW Q_HIGH\n"
            "Sample from a uniform distribution with lower bound\n"
            "Q_LOW and upper bound Q_HIGH\n";
        return 1;
    }

    double q_low = std::stod(argv[1]);
    double q_high = std::stod(argv[2]);

    // Seed random number generator
    unsigned seed =
        std::chrono::system_clock::now().time_since_epoch().count();
    std::mt19937_64 generator(seed);

    // Sample from uniform distribution
    std::uniform_real_distribution<double> distribution(q_low, q_high);
    double q_sampled = distribution(generator);

    // Print sampled parameter
    std::cout.precision(17);
    std::cout << q_sampled << std::endl;

    return 0;
}
